### PR TITLE
rrdp-iostat: use scanf instead of Str.regexp

### PR DESCRIPTION
**Pull request by Wescoeur**
Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>

### CODE REVIEW

**Review by lindig — APPROVED** [review]
Thanks for your submission. This looks good to me. It is good to get rid of `Str` because it is unreliable and `Re` would be a better replacement. Using `scanf` here is fine, too.

File: src/rrdp-iostat/rrdp_iostat.ml [hunk 1]
```diff
@@ -297,16 +291,12 @@ let exec_tap_ctl () =
...
-      begin
-        Printf.printf "\"%s\" returned a line that could not be parsed. Ignoring.\n" tap_ctl;
-        Printf.printf "Offending line: %s\n" str;
-        None
-      end
-    else extract_groups str
+    try
+      Scanf.sscanf str "pid=%d minor=%d state=%s args=%s@:%s" extract_vdis
+    with Scanf.Scan_failure _ | Failure _ | End_of_file ->
+      D.warn {|"%s" returned a line that could not be parsed. Ignoring.\n|} tap_ctl;
```
Comment (lindig) [inline src/rrdp-iostat/rrdp_iostat.ml:297]: These log lines should not have a trailing `\n`

**Comment by psafont** [comment]
This is a fix for the issue found in https://github.com/xapi-project/blktap/pull/345

This is, state was parsed as a series of decimal digits when it can be a hexadecimal number, and after that the result is ignored.

**Review by psafont — APPROVED** [review]

**Review by lindig — CHANGES_REQUESTED** [review]
Please remove the `\n` from the log lines.

File: src/rrdp-iostat/rrdp_iostat.ml [hunk 2]
```diff
@@ -279,11 +276,8 @@ let refresh_phypath_to_sr_vdi () =
 
 let exec_tap_ctl () =
   let tap_ctl = "/usr/sbin/tap-ctl list" in
-  let extract_groups str =
-    let pid = int_of_string (Str.matched_group 1 str)
-    and minor = int_of_string (Str.matched_group 2 str)
-    and phypath = Str.matched_group 4 str
-    in
+  let extract_vdis pid minor state kind phypath =
+    if not (kind = "vhd" || kind = "aio") then raise (Failure "Unkown type") ;
```
Comment (lippirk) [inline src/rrdp-iostat/rrdp_iostat.ml:280]:
```suggestion
    if not (kind = "vhd" || kind = "aio") then raise (Failure "Unknown type") ;
```

but this string is never printed, so this is not a problem

**Review by lippirk — APPROVED** [review]

**Review by lindig — APPROVED** [review]

**Comment by lindig** [comment]
Thanks. Will merge it after the CI did its job.

**Comment by Wescoeur** [comment]
Nice! Thank you. :)

**Pull request merged**